nemo_skills/inference/server/reward_model.py

```diff
@@ -118,10 +118,28 @@ def __init__(self, **kwargs):
         self.model = model_list.data[0].id
 
     def _score_single_prompt(self, prompt):
-        response = requests.post(self.request_url, json={"input": prompt, "model": self.model})
-        per_token_scores = response.json()['data'][0]['data']
-        last_token_score = per_token_scores[-1]
+        """Score a single prompt"""
+
+        per_token_scores = None
+        try:
+            response = requests.post(self.request_url, json={"input": prompt, "model": self.model})
+            output = response.json()
+            per_token_scores = output['data'][0]['data']
+        except requests.exceptions.HTTPError as err:
+            print(f"Request failed: {err}")
```

Comment:
Please use LOG.warning instead

Reply:
I think it would be good to add the error as an explicit field in the output, maybe with `inference_error` field or something similarly named. That way it will be easier to understand why the score is 0 and it increases the chance that users notice that perhaps a lot of their requests have failed (not everyone is checking the logs)

Reply:
Made the suggested changes and ran the GPU test locally. 
Will commit the change if the CPU tests pass